redash/handlers/data_sources.py

```diff
@@ -34,12 +34,19 @@ def get(self):
 
 
 class DataSourceResource(BaseResource):
-    @require_admin
     def get(self, data_source_id):
-        data_source = models.DataSource.get_by_id_and_org(
-            data_source_id, self.current_org
+        data_source = get_object_or_404(
+            models.DataSource.get_by_id_and_org, data_source_id, self.current_org
+        )
+        require_access(data_source, self.current_user, view_only)
+        ds = (
+            data_source.to_dict(all=self.current_user.has_permission("admin"))
+            if self.current_user.has_permission("list_data_sources")
+            else {}
+        )
```

Comment:
Please correct me if I misunderstand something:

1. for admins it will serialize all fields + `view_only` flags
2. for those who can `list_data_sources` it will serialize limited set of fields + `view_only` flags
3. for all others it will return just `view_only` flags

If so - it looks good. Maybe just add some comment to explain this logic, or split this one-liner to make it a bit easier to understand.

Reply:
@kravets-levko correct, sorry if the code is a bit weird, for now I only made de simplest changes to what was there 😅

If @arikfr agrees with this flow I'll happily try to simplify the code (or add the explaining comment) + add tests to make sure it keeps working as it should.

Reply:
I guess this flow is fine, but we still need the relevant UI changes (hide data source if we don't have its name/type).

Reply:
I've added tests and updated the code.

One extra thing to @kravets-levko flow that was not mentioned: The user has to either be an admin or have at least view permissions to follow to that flow, otherwise a 403 is returned.